### ISSUE TITLE
Analytics or custom head tag

### ISSUE THREAD

**Issue by XinYaanZyoy** [KosmosisDire/obsidian-webpage-export#153]
an ability to add analytics js files, and in general any custom head tag parts.

this is actually a question and an enhancement request if the answer is positive.

**Comment by Thewest123** [comment]
I had the same idea and solved it by modifying the exported `libs/script/webpage.js` file. Include this at the bottom of the file and change the `window.onload` function:

```javascript
/*This function will load script and call the callback once the script has loaded*/
function loadScriptAsync(scriptSrc, callback)
{
    if (typeof callback !== 'function')
    {
        throw new Error('Not a valid callback for async script load');
    }
    var script = document.createElement('script');
    script.onload = callback;
    script.src = scriptSrc;
    document.head.appendChild(script);
}

function loadAnalytics()
{
    /* This is the part where you call the above defined function and "call back" your code which gets executed after the script has loaded */
    loadScriptAsync('https://www.googletagmanager.com/gtag/js?id=G-XXXXXXXXXX', function()
    {
        window.dataLayer = window.dataLayer || [];
        function gtag(){dataLayer.push(arguments);}
        gtag('js', new Date());

        gtag('config', 'G-XXXXXXXXXX');
        console.log("Analytics loaded");
    });
}

function loadHotjar()
{
    (function(h,o,t,j,a,r){
        h.hj=h.hj||function(){(h.hj.q=h.hj.q||[]).push(arguments)};
        h._hjSettings={hjid:XXXXXXX,hjsv:6};
        a=o.getElementsByTagName('head')[0];
        r=o.createElement('script');r.async=1;
        r.src=t+h._hjSettings.hjid+j+h._hjSettings.hjsv;
        a.appendChild(r);
    })(window,document,'https://static.hotjar.com/c/hotjar-','.js?sv=');
    console.log("Hotjar loaded");
}

window.onload = function()
{
    if (window.location.protocol == "file:") initializeForFileProtocol();
    initializePage(document);
    loadAnalytics();
    loadHotjar();
}
```

**Comment by XinYaanZyoy** [comment]
that's a good solution. but unfortunately, it doesn't help me. because I export the same vault periodically, and the plugin modifies the `libs/script/webpage.js` file on each export. 

@KosmosisDire, if I understand it right, `libs/script/webpage.js` file doesn't really need to be modified on each export, does it?

The alternate solution I see, as I have hinted already, is to have a way to provide external snippets to be adjoined to `libs/script/webpage.js` file on each export.

**Comment by KosmosisDire** [comment]
The webpage.js file should not be modified on each export. In the past it was modified every time obsidian was reloaded, however that is no longer the case in the latest versions.

I do plan to add this ass a built-in feature in the future though.

**Comment by XinYaanZyoy** [comment]
umm... I'm using v1.6.0. I checked it again, it DOES modify the webpage.js file for me. Do you mean the latest commits have solved this? if so, how do I update it in my vault? 

also, does Obsidian update it automatically when you tag a new version?

**Comment by KosmosisDire** [comment]
Sorry I misunderstood the message and got mixed up. Yes you are right that the file is modified on export! This problem would be solved by this feature requesst: #152, but of course the best solution would be to add a specific way to include custom header html which is what I plan to do.

**Comment by KosmosisDire** [comment]
This feature has been added in 1.8.0! :)
If you find any issues please create a new issue
Thanks!
